docs/_deployments/serverless-framework.md

```diff
@@ -0,0 +1,342 @@
+
+---
+
+### Set up AWS Lambda
+
+[AWS Lambda][aws-lambda] is a serverless, function-as-a-service (FaaS) platform that allows you to run code without managing servers. In this section, we'll configure your local machine to access AWS Lambda.
+
+> 💡 Skip this section if you have [configured a profile][aws-profiles] on your local machine with access to AWS Lambda.
+
+**1. Sign up for an AWS account**
+
+First off, you'll need an AWS account. If you already an account, then you can go to the next step.
+
+To create an account, [sign up for AWS](https://aws.amazon.com/) and follow the steps.
+
+> 💡 You may be asked for payment information during the sign up. Don't worry, this guide only uses the [free tier][aws-pricing].
+
+**2. Create an AWS access key**
+
+Next, you'll need programmatic access to the AWS account to deploy onto AWS Lambda. In the world of AWS, this requires an **Access Key ID** and **Secret Access Key**.
+
+We recommend watching this short, step-by-step video to 🍿 [create an IAM user and download the access keys](https://www.youtube.com/watch?v=KngM5bfpttA).
+
+> 💡 **Do you already have an IAM user?** Follow this official guide to [create access keys for an existing IAM user][aws-iam-user].
+
+**3. Install the AWS CLI**
+
+The AWS tools are available as a Command Line Interface (CLI) and can be [installed on macOS, Windows, or Linux][aws-cli-install].
+
+On macOS, you can install the AWS CLI by [downloading the latest package installer](https://awscli.amazonaws.com/AWSCLIV2.pkg).
+
+**4. Configure an AWS profile**
+
+You can use the AWS CLI to configure a profile that stores your access key pair on your local machine. This profile is used by the CLI and other tools to access AWS.
+
+The quickest way to [configure your profile][aws-cli-configure] is to run this command and follow the prompts:
+
+```zsh
+aws configure
+# AWS Access Key ID [None]: <your-aws-access-key>
+# AWS Secret Access Key [None]: <your-aws-secret>
+# Default region name [None]: us-east-1
+# Default output format [None]: json
+```
+
+> 💡 Customize the [region][aws-cli-region] and [output format][aws-cli-output-format] best for you.
+
+That wraps up configuring your local machine to access AWS. 👏 Next, let's do the same with the Serverless Framework.
+
+---
+
+### Set up Serverless Framework
+
+Now we can set up the free [Serverless Framework](https://www.serverless.com/open-source/) open source tools on your local machine. These tools allow you to easily configure, debug, and deploy your app to AWS Lambda.
+
+**1. Install the Serverless Framework CLI**
+
+The Serverless tools are available as a Command Line Interface (CLI) and can be [installed on macOS, Windows, or Linux](https://www.serverless.com/framework/docs/getting-started/). You can install it using npm with the command:
+
+```shell
+npm install --save-dev serverless
+```
```

Comment:
I've chosen to install the Serverless CLI as a developer dependency instead of globally. It's an approach that we used in [slackapi/bolt-examples-aws-re-invent-2020](https://github.com/slackapi/bolt-examples-aws-re-invent-2020/) and I like how this cuts down on the system installs that we ask of our community.

It also makes the CLI more portable for the other Bolt implementations, such as how we bundle the Serverless CLI with the Python example: https://github.com/slackapi/bolt-examples-aws-re-invent-2020/tree/main/api-gateway-lambda/python#4-deploy-to-aws-lambda

Thoughts?

Reply:
Yeah, this makes sense.